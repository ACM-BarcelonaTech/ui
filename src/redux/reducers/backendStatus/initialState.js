const initialExperimentBackendStatus = {
  loading: false,
  error: false,
  status: {
  },
};

const initialState = {};

export default initialState;

export { initialExperimentBackendStatus };
