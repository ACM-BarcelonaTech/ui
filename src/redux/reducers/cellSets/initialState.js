const initialState = {
  properties: {},
  hierarchy: [],
  selected: [],
  loading: true,
  error: false,
};

export default initialState;
