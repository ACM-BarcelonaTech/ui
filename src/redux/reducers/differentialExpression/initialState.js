const initialState = {
  properties: {
    data: [],
    comparisonGroup: {},
    comparisonType: null,
    loading: false,
    error: false,
    total: 0,
  },
  comparison: {
    type: 'within',
    advancedFilters: [],
    group: {
      between: {
        cellSet: null,
        compareWith: null,
        basis: null,
      },
      within: {
        cellSet: null,
        compareWith: null,
        basis: null,
      },
    },
    ordering: {
      orderBy: 'logFC',
      orderDirection: 'DESC',
    },
  },
};

export default initialState;
