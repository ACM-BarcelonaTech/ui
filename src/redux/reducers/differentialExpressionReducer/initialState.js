const initialState = {
  properties: {
    data: {},
    cellSets: {},
    loading: false,
    error: false,
  },
};

export default initialState;
