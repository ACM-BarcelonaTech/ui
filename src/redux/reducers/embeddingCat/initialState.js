const initialState = {
  width: 700,
  height: 550,
  pointSize: 5,
  toggleInvert: '#FFFFFF',
  masterColour: '#000000',
  umap1Domain: null,
  umap2Domain: null,
  titleText: '',
  titleSize: 20,
  titleAnchor: 'start',
  axisTitlesize: 13,
  axisTicks: 13,
  transGrid: 0,
  axesOffset: 10,
  masterFont: 'sans-serif',
  xaxisText: 'UMAP 1',
  yaxisText: 'UMAP 2',
  pointStyle: 'circle',
  pointOpa: 5,
  g1Color: 'red',
  g2mColor: 'green',
  sColor: 'blue',
  legendTextColor: '#FFFFFF',
  legendEnabled: true,
  legend: null,
  geneexpLegendloc: '',
  labelSize: 28,
  labelShow: 1,
  labelFont: 2,
  labelsEnabled: true,
  selectedClusters: [],
  testVar: null,
};

export default initialState;
