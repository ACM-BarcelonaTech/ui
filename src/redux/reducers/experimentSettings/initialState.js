const metaInitialState = {
  loading: false,
  completingStepError: false,
  loadingSettingsError: false,
  selectedConfigureEmbeddingPlot: 'cellCluster',
  changedQCFilters: new Set(),
};

const initialState = {
  info: {
    experimentId: null,
    experimentName: null,
    sampleIds: [],
    paramsHash: '',
  },
  processing: {
    meta: metaInitialState,
  },
  originalProcessing: {},
};

export { metaInitialState };
export default initialState;
