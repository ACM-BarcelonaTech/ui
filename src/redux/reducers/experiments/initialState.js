const experimentTemplate = {
  projectUuid: null,
  name: null,
  description: 'Analysis description',
  id: null,
  createdDate: null,
  lastViewed: null,
  notifyByEmail: false,
  meta: { organism: null, type: '10x' },
  sampleIds: [],
};

const initialState = {
  meta: {
    loading: false,
    error: false,
    saving: false,
  },
  ids: [],
};

export default initialState;
export { experimentTemplate };
