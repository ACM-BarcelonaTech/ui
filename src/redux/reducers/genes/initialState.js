/**
 * The store has the following format:
 * {
 *    genes: {
 *      properties: {
 *        loading: [],
 *        error: false,
 *        data {
  *        TGFB1: {
 *            id: "ENS013131313",
 *            dispersion: 0.2
 *         }
 *        },
 *        views: {
 *          gene-view-props: {
 *            fetching: false,
 *            error: false,
 *            data: ['TGFB1', 'FOXP2', 'MYC'],
 *            total: 20
 *          }
 *        }
 *      }
 *      expression: {
 *        data: {
 *          TGFB1: {
 *            min: 0.0,
 *            max: 7.0,
 *            data: [3.4, 2.2, 1.1]
 *          }
 *        },
 *        error: false,
 *        loading: [],
 *      },
 *      selected: [],
 *      focused: "MYC"
 *    }
 * }
 */

const initialViewState = {
  fetching: false,
  error: false,
  data: [],
};

const initialExpressionState = {
  min: 0,
  max: 0,
  data: [],
};

const initialState = {
  properties: {
    loading: [],
    views: {},
    data: {},
  },
  expression: {
    loading: [],
    error: false,
    data: {},
  },
  selected: [],
  focused: undefined,
};

export { initialViewState, initialExpressionState };
export default initialState;
