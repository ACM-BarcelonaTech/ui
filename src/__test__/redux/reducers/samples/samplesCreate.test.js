import samplesCreateReducer from 'redux/reducers/samples/samplesCreate';

import initialState, { sampleTemplate } from 'redux/reducers/samples/initialState';

const projectUuid = 'projectUuid';

const newSample = {
  ...sampleTemplate,
  name: 'sampleName',
  uuid: 'uuid',
  projectUuid,
  type: '10x',
};

describe('samplesCreate', () => {
  it('returns correct state starting from initial state', () => {
    const newState = samplesCreateReducer(initialState, { payload: { sample: newSample } });

    expect(newState).toMatchSnapshot();
  });

  it('returns correct state starting from a preexisting sample', () => {
    const anotherSample = {
      ...sampleTemplate,
      name: 'oldSampleName',
      uuid: 'oldUuid',
      projectUuid,
      type: '10x',
    };

    const anotherSampleState = {
      ...initialState,
      [anotherSample.uuid]: anotherSample,
    };

    const newState = samplesCreateReducer(anotherSampleState, { payload: { sample: newSample } });

    expect(newState).toMatchSnapshot();
  });
});
