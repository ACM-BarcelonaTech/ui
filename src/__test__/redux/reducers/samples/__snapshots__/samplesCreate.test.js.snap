// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`samplesCreate returns correct state starting from a preexisting sample 1`] = `
Object {
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": false,
  },
  "oldUuid": Object {
    "complete": false,
    "createdDate": null,
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": null,
    "metadata": Object {},
    "name": "oldSampleName",
    "projectUuid": "projectUuid",
    "species": null,
    "type": "10x",
    "uuid": "oldUuid",
  },
  "uuid": Object {
    "complete": false,
    "createdDate": null,
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": null,
    "metadata": Object {},
    "name": "sampleName",
    "projectUuid": "projectUuid",
    "species": null,
    "type": "10x",
    "uuid": "uuid",
  },
}
`;

exports[`samplesCreate returns correct state starting from initial state 1`] = `
Object {
  "meta": Object {
    "error": false,
    "loading": false,
    "saving": false,
  },
  "uuid": Object {
    "complete": false,
    "createdDate": null,
    "error": false,
    "fileNames": Array [],
    "files": Object {},
    "lastModified": null,
    "metadata": Object {},
    "name": "sampleName",
    "projectUuid": "projectUuid",
    "species": null,
    "type": "10x",
    "uuid": "uuid",
  },
}
`;
